api/src/main/java/jakarta/json/spi/JsonProvider.java

```diff
@@ -477,4 +477,16 @@ public JsonNumber createValue(BigDecimal value) {
     public JsonNumber createValue(BigInteger value) {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * Creates a JsonNumber.
+     *
+     * @param value a JSON number
+     * @return the JsonNumber for the number
+     *
+     * @since 2.1
+     */
+    public JsonNumber createValue(Number value) {
+        throw new UnsupportedOperationException();
```

Comment:
would it make sense to have the default impl which would just call `createValue(value.intValue())` for int, `createValue(value.doubleValue())` for double etc? In any case, the parameter name should be renamed to sth more meaningful, what about `number`?

Reply:
That is more or less how I solved it in the implementation:
https://github.com/eclipse-ee4j/parsson/pull/30/files

From my point of view, it makes more sense to do it in the API if we are able to cover every scenario. Otherwise it will be half done.

There is one scenario  where Number is not known, for example, if somebody implements its own Number. In this case I think we cannot reuse any existing method easily. What is your opinion?.

Regarding the parameter name, you mean to replace 'value' by 'number', right?.

Reply:
> if somebody implements its own Number

handle known/simple cases and throw UOE for unknowns? I agree it does not cover every scenario but even ~75% at this level is IMO good enough.

> Regarding the parameter name, you mean to replace 'value' by 'number', right?.

yes

Reply:
I have updated it with the supported types and the 'value' was changed to 'number'.

Reply:
could the exception contain some message, please?

Reply:
Done